modules/renter/uploadheap.go

```diff
@@ -236,6 +238,30 @@ func (r *Renter) managedBuildChunkHeap(hosts map[string]struct{}) {
 	// the heap.
 	id := r.mu.Lock()
 	for _, file := range r.files {
+		// check for local file
+		tf, exists := r.persist.Tracking[file.name]
+		if exists {
+			// Build maps that map contract id to its offline and goodForRenew
+			// status
+			contractIDs := make(map[types.FileContractID]struct{})
+			for cid := range file.contracts {
```

Comment:
When accessing fields of the file like `file.name` and `file.contracts` you need to lock `file.mu`. You can find an example in `FileList`.